### PR TITLE
Complete Home Screen UI, Networking and pagination.

### PR DESCRIPTION
**Pull request by MohamedSaiko**
Code review is required.

### CODE REVIEW

**Comment by MohamedSaiko** [comment]
Authentication, SignIn Screen are not finished yet.
large commits because I splitted the old project into 2 projects.

File: Food-Advisor-StoryBoard/Home/HomeViewController.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+//
+//  Created by Mohamed Sayed on 07.04.23.
+//
+
+import UIKit
+
+final class HomeViewController: UIViewController {
+    private var homeViewModel = HomeViewModel()
+    private let searchController = UISearchController()
+    private let refreshControl = UIRefreshControl()
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Home/HomeViewController.swift:14]: please add empty new line here to be more readable

File: Food-Advisor-StoryBoard/Home/HomeViewController.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @IBOutlet private weak var postsTableView: UITableView!
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Home/HomeViewController.swift:20]: also here please add empty new line here to be more readable

File: Food-Advisor-StoryBoard/Home/HomeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        navigationItem.leftBarButtonItem = UIBarButtonItem(image: logo)
+        navigationItem.leftBarButtonItem?.isEnabled = false
+        navigationItem.rightBarButtonItem = UIBarButtonItem(image: search, style: .plain, target: self, action: #selector(searchTapped))
+        refreshControl.addTarget(self, action: #selector(handleRefreshControl), for: .valueChanged)
+        postsTableView.addSubview(refreshControl)
+    }
+    private func updateTableView() {
+        homeViewModel.getPosts { [weak self] in
+            DispatchQueue.main.async {
+                self?.postsTableView.reloadData()
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Home/HomeViewController.swift]: you should guard self here

File: Food-Advisor-StoryBoard/Home/HomeViewController.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+extension HomeViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return homeViewModel.posts.count
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let post = homeViewModel.posts[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: postsCellIdentifier, for: indexPath) as? PostCell
+        cell?.configure(post: post)
+        homeViewModel.tableViewWillEnd(indexPath: indexPath) {
+            self.postsTableView.tableFooterView = createFooterSpinner()
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Home/HomeViewController.swift]: you don't need to use `self` here

File: Food-Advisor-StoryBoard/Home/HomeViewController.swift [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+}
+// MARK: - Data Source
+extension HomeViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return homeViewModel.posts.count
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let post = homeViewModel.posts[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: postsCellIdentifier, for: indexPath) as? PostCell
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Home/HomeViewController.swift:93]: you should guard `cell` here

File: Food-Advisor-StoryBoard/Home/PostCell.swift [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+//  PostCell.swift
+//  Food-Advisor-StoryBoard
+//
+//  Created by Mohamed Sayed on 02.12.23.
+//
+
+import UIKit
+
+final class PostCell: UITableViewCell {
+    @IBOutlet private weak var postBody: UILabel!
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Home/PostCell.swift:11]: you could rename it to bodyLabel. WDYT?

File: Food-Advisor-StoryBoard/Models/User.swift [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+//  User.swift
+//  Food Advisor
+//
+//  Created by Mohamed Sayed on 24.03.23.
+//
+
+import Foundation
+
+struct User: Decodable {
+    let username: String
```
Comment (akabel92) [inline Food-Advisor-StoryBoard/Models/User.swift:11]: userName

**Review by akabel92 — APPROVED** [review]

**Pull request merged**